MiscGlowMasks.cs

```diff
@@ -9,53 +10,64 @@ namespace GoldensMisc
 {
 	public static class MiscGlowMasks
 	{
-		public static short UndyingSpear;
-		public static short UndyingSpearProjectile;
-		public static short RodofWarping;
-		const short Count = 3;
-		static short End;
+		public static Texture2D UndyingSpear;
+		public static Texture2D UndyingSpearProjectile;
+		public static Texture2D RodofWarping;
 		public static bool Loaded;
 		
 		public static void Load()
 		{
-			Array.Resize(ref TextureAssets.GlowMask, TextureAssets.GlowMask.Length + Count);
-			
-			short i = (short)(TextureAssets.GlowMask.Length - Count);
-			TextureAssets.GlowMask[i] = ModContent.Request<Texture2D>("GoldensMisc/Items/Weapons/UndyingSpear_Glow");
-			UndyingSpear = i;
-			i++;
-			TextureAssets.GlowMask[i] = ModContent.Request<Texture2D>("GoldensMisc/Projectiles/UndyingSpear_Glow");
-			UndyingSpearProjectile = i;
-			i++;
-			TextureAssets.GlowMask[i] = ModContent.Request<Texture2D>("GoldensMisc/Items/Tools/RodofWarping_Glow");
-			RodofWarping = i;
-			i++;
-			End = i;
+			if (!Loaded)
+            {
+				UndyingSpear = ModContent.Request<Texture2D>("GoldensMisc/Items/Weapons/UndyingSpear_Glow", ReLogic.Content.AssetRequestMode.ImmediateLoad).Value;
+				UndyingSpearProjectile = ModContent.Request<Texture2D>("GoldensMisc/Projectiles/UndyingSpear_Glow", ReLogic.Content.AssetRequestMode.ImmediateLoad).Value;
+				RodofWarping = ModContent.Request<Texture2D>("GoldensMisc/Items/Tools/RodofWarping_Glow", ReLogic.Content.AssetRequestMode.ImmediateLoad).Value;
+				Loaded = true;
+			}
 			
-			Loaded = true;
 		}
 		
 		public static void Unload()
 		{
 			if(Loaded)
 			{
-				//Remove our glow masks so the mod can unload properly(?)
-				TextureAssets.GlowMask[UndyingSpear] = null;
-				TextureAssets.GlowMask[UndyingSpearProjectile] = null;
-				TextureAssets.GlowMask[RodofWarping] = null;
-
-				//Shrink array back to original if possible
-				if(TextureAssets.GlowMask.Length == End)
-				{
-					Array.Resize(ref TextureAssets.GlowMask, TextureAssets.GlowMask.Length - Count);
-				}
-
+				UndyingSpear = null;
+				UndyingSpearProjectile = null;
+				RodofWarping = null;
 				Loaded = false;
-				UndyingSpear = 0;
-				UndyingSpearProjectile = 0;
-				RodofWarping = 0;
-				End = 0;
 			}
 		}
+
+	}
+
+	public abstract class GlowingModItem : ModItem
+    {
+		private Texture2D _glowMask;
```

Comment:
My C# skills are getting rusty and I can't tell why a public setter and getter is useful.

Reply:
Technically it is more compatible for some types of binaries but I can just make glowMask public, I have been working on various projects on bepinex where it is more common since those are designed as librarys. I just wrote that way, It isn't necessary and it is not stylistic for this project.

Ill switch to just a public field.